File: Frontend/src/interfaces/v12/location/location.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface LocationDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface LocationCreateUpdateDto {
  name: string;
}

export interface LocationSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/Location/Location.cs
 using Wms.Api.Entities; 

namespace Wms.Api.Dto.Location
{
    public class LocationDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class LocationCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class LocationSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
    
    public class LocationFindByParametersDto : PagedRequestAbstractDto
    {
        public Guid[] LocationIds { get; set; } = [];
    }

    public class ActiveLocationDto  
    {
        public Guid LocationId { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
    }
}
